core/server/master/src/main/java/alluxio/master/metastore/CachingInodeStore.java

```diff
@@ -11,40 +11,45 @@
 
 package alluxio.master.metastore;
 
-import java.util.Iterator;
-import java.util.Optional;
-
-import com.google.common.cache.Cache;
-import com.google.common.cache.CacheBuilder;
-
 import alluxio.Configuration;
 import alluxio.PropertyKey;
+import alluxio.conf.InstancedConfiguration;
 import alluxio.master.file.meta.Edge;
 import alluxio.master.file.meta.Inode;
 import alluxio.master.file.meta.InodeDirectoryView;
 import alluxio.master.file.meta.InodeView;
 import alluxio.master.file.meta.MutableInode;
 
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+import java.util.Iterator;
+import java.util.Optional;
+
 /**
  * An inode store which caches inode tree metadata and delegates to another inode store for cache
  * misses.
  */
 public final class CachingInodeStore implements InodeStore {
   private final InodeStore mBackingStore;
+  private final InstancedConfiguration mConf;
 
   // Cache recently-accessed inodes.
-  private final Cache<Long, MutableInode<?>> mInodeCache = CacheBuilder.newBuilder()
-      .maximumSize(Configuration.getInt(PropertyKey.MASTER_METASTORE_INODE_CACHE_SIZE)).build();
+  private final Cache<Long, MutableInode<?>> mInodeCache;
 
   // Cache recently-accessed inode tree edges.
-  private final Cache<Edge, Long> mEdgeCache = CacheBuilder.newBuilder()
-      .maximumSize(Configuration.getInt(PropertyKey.MASTER_METASTORE_INODE_CACHE_SIZE)).build();
+  private final Cache<Edge, Long> mEdgeCache;;
 
   /**
    * @param backingStore the backing inode store
    */
-  public CachingInodeStore(InodeStore backingStore) {
+  public CachingInodeStore(InodeStore backingStore, InstancedConfiguration conf) {
```

Comment:
Add another constructor without the instanced conf and make this one for testing only? Or do you imagine using instanced conf more on the server side?

Reply:
I think we should eventually do away with `Configuration.global` and use instanced configuration everywhere for consistency and testability. To make it easier in the future, I like pushing the call to `Configuration.global()` as high in the call stack as possible